src/vs/workbench/contrib/editSessions/browser/editSessionsWorkbenchService.ts

```diff
@@ -173,30 +177,45 @@ export class EditSessionsWorkbenchService extends Disposable implements IEditSes
 			return true;
 		}
 
+		const authenticationSession = await this.getAuthenticationSession();
+		if (authenticationSession !== undefined) {
+			this.#authenticationInfo = authenticationSession;
+			this.storeClient.setAuthToken(authenticationSession.token, authenticationSession.providerId);
+		}
+
+		return authenticationSession !== undefined;
+	}
+
+	private async getAuthenticationSession() {
 		// If the user signed in previously and the session is still available, reuse that without prompting the user again
-		const existingSessionId = this.existingSessionId;
-		if (existingSessionId) {
-			this.logService.trace(`Searching for existing authentication session with ID ${existingSessionId}`);
-			const existing = await this.getExistingSession();
-			if (existing !== undefined) {
-				this.logService.trace(`Found existing authentication session with ID ${existingSessionId}`);
-				this.#authenticationInfo = { sessionId: existing.session.id, token: existing.session.idToken ?? existing.session.accessToken, providerId: existing.session.providerId };
-				this.storeClient.setAuthToken(this.#authenticationInfo.token, this.#authenticationInfo.providerId);
-				return true;
+		if (this.existingSessionId) {
```

Comment:
I could be incorrect, but think this check is unnecessary since `this.getExistingSession` checks against `ExistingSession`s that always have defined ids.

Reply:
`this.existingSessionId` is an authentication session id stored in memento storage which indicates the user's preferred auth session for settings sync on the current machine, so it's not guaranteed to be defined (e.g. if the user has never signed into edit sessions before). If it doesn't exist, we can bypass calling `this.getExistingSession()` since we know that method is going to return `undefined`.